python/GafferDelightUI/ShaderMenu.py

```diff
@@ -71,35 +71,158 @@ def __shaderSubMenu( oslPath ) :
-					subMenu = "Texture 3D"
-				elif tag == "utility" :
-					subMenu = "Utility"
-				elif tag == "toon" :
-					subMenu = "Toon"
-				else :
-					subMenu = "Other"
+				subMenu = tag.capitalize()
 
-			if subMenu is None :
+			if shaderFileName == "distantLight" :
+				continue
+			elif shaderFileName == "directionalLight" :
+				continue
+			elif shaderFileName == "distantLight" :
+				continue
+			elif shaderFileName == "pointLight" :
+				continue
+			elif shaderFileName == "spotLight" :
+				continue
+			elif shaderFileName == "environmentLight" :
+				continue
+			elif shaderFileName == "displacementShader" :
+				continue
+			elif shaderFileName == "surfaceShader" :
+				continue
+			elif shaderFileName == "plusMinusAverage" :
+				continue
+			elif shaderFileName == "addDoubleLinear" :
 				continue
+			elif shaderFileName == "multDoubleLinear" :
+				continue
+			elif shaderFileName == "tripleShadingSwitch" :
+				continue
+			elif shaderFileName == "pfx" :
+				continue
+			elif shaderFileName == "particleCloud" :
+				continue
+			elif shaderFileName == "shadingEngine_surface" :
+				continue
+			elif shaderFileName == "surfaceLuminance" :
+				continue
+			elif shaderFileName == "filter_multiply" :
+				continue
+			elif shaderFileName == "lens_distortion" :
+				continue
+			elif shaderFileName == "dl3DelightMaterial" :
+				continue
+			elif shaderFileName == "transparent" :
+				continue
+			elif shaderFileName == "shadingEngine_displacement" :
+				subMenu = "Displacement"
+			elif shaderFileName == "builtinDisplacement" :
+				subMenu = "Displacement"
+			elif shaderFileName == "glassInterior" :
+				subMenu = "Surface"
+			elif shaderFileName == "bump2d" :
+				subMenu = "Bump"
+			elif shaderFileName == "bump3d" :
+				subMenu = "Bump"
+			elif shaderFileName == "dlPrimitiveAttribute" :
+				subMenu = "Utility"
+			elif shaderFileName == "dlAOVGroup" :
+				subMenu = "AOV"
+			elif shaderFileName == "dlAOVGroupFour" :
+				subMenu = "AOV"
+			elif shaderFileName == "hsvToRgb" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "rgbToHsv" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "blendColors" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "Curve UV Coordinates" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "uvCoord" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "contrast" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "remapHsv" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "remapValue" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "shadingEngine_surface_switch" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "shadingEngine_displacement_switch" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "uvChooser" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "multiplyDivide" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "Set Range" :
```

Comment:
I think this should be "setRange"? It appears in "Utility" now, instead of "Utility (Legacy)".

Reply:
Actually there are two set range shaders - dlSetRange.oso and setRange.oso. One operates on float values and the other one on vectors. I changed the Gaffer name of the float shader to Set Range Float, kept the other one as Set Range and placed both in the main Utility menu.